api/python/quilt3/packages.py

```diff
@@ -880,10 +914,49 @@ def set(self, logical_key, entry=None, meta=None):
             entry = PackageEntry([url], size, None, orig_meta)
         elif isinstance(entry, PackageEntry):
             entry = entry._clone()
+
+        elif object_is_serializable(entry):
+            # Use file extension from serialization_location, fall back to file extension from logical_key
```

Comment:
Using the file extension here seems a bit fragile, especially since there are potentially two (for both the logical and physical keys). Since this is really a convenience mechanism, I'm not sure how important it is to support multiple serialization methods. Users can manually serialize then add the file if they want to control serialization.

Reply:
We haven't previously required that logical keys have extensions. If we make that change here, we should be sure to do it intentionally and document it.

Reply:
The key thing I want to accomplish with this piece of code is preventing `pkg.set('mydataframe.csv', df)` from doing something stupid like serializing the dataframe as parquet. We aren't requiring the user to give us a file extension, we are just respecting it if is is specified, since it looks like it is not uncommon for logical keys to have extensions.

I added an exception if the user gives conflicting file extensions.

Reply:
As written in this PR (selecting the serializer from the file extension) is correct and true to the user's intention. We are actually moving away from a worse problem that we had before: we would _always_ select parquet for dataframes, irrespective of the user's choice of file extension. Proposed logic is strictly better.